### DISCUSSION TITLE
Welcome to esphome-dsckeybus Discussions!

### DISCUSSION THREAD

**Discussion by Dilbert66** [Dilbert66/esphome-dsckeybus#21] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by neohusky** [comment]
Hi Alain

Thank you for your work on this project. I'm a recent convert from Openhab and I'm still getting my head around Home Assistant.

I've flashed my wemos D1 mini successfull and I have got it connected and working with my dsc alarm.  I've tested arm and disarm successfully using the Developer Tools / Services tool.

I'm being lazy, but, is there a guide or can you provide config files/ a card to integrate this into HA? 

Ultimately, I'd love to integrate/expose this into homekit as a security device.  Can you see that being possible?

Thanks in advance,
Theo

**Comment by Dilbert66** [comment]
This component is built around esphome which only works on home assistant at this time. For home kit you might want to look at using this component on the dsckeybusinterface library home project at: 
https://github.com/taligentx/dscKeybusInterface/tree/master/examples/esp8266/Homebridge-MQTT

**Reply by neohusky** [reply]
Thanks for pointing me to the homebridge mqtt implementation. I had seen that previously,but, wanted to try an avoid an additional broker to learn.  

A couple of questions:
1. Since I'm new to home assistant and really want to give it a go as a replacement for openhab do you have an example configuration with card for home your esphome project. 

2. Once I have these all the sensors, statuses and switches implemented into home assistant I was looking to use HomeKit integration to expose and control these with HomeKit also. Do you think this is possible (even though it may seem like extra work/duplication)?

**Comment by Dilbert66** [comment]
1. There is an example setup in the project readme. https://github.com/Dilbert66/esphome-dsckeybus
2.  I don't know anything about homekit sorry.

**Reply by neohusky** [reply]
A quick update...After setting the config and confirming everything worked and I could arm and disarm the panel from  the dashboard I enabled the HomeKit integration which automatically did what I wanted in HomeKit :)

Thanks again!!

**Comment by younesetre** [comment]
Hello
I am new in homeassistant.  the esp8266 I have to place it in parallel with the dsc or do I have to remove the dsc?

**Comment by Dilbert66** [comment]
In parallel. It emulates a keypad in order to allow homeassistant to talk with the dsc.

**Reply by younesetre** [reply]
Thanks,

The  diagram of my alarm is little different. I can't find  yellow or green wire but i have 3 wire : data, v_key and G
![alarm unit](https://user-images.githubusercontent.com/6561267/135907055-bf09b68f-5b5c-4625-a50b-0aefa1711851.jpg)
![keypad](https://user-images.githubusercontent.com/6561267/135907172-ce85f1a7-9e7c-4a47-a793-8c97b45343e3.jpg)

**Reply by younesetre** [reply]
I guess wire data is the green wire and v_key is the yellow wire

**Comment by Dilbert66** [comment]
What model of alarm system is this? The bus you show is not compatible with  this library or any that I'm aware of.

**Comment by younesetre** [comment]
https://homedocbox.com/118732531-Environmental_Safety/Reading-carefully-this-guide-operating-this-system-strictly-on-the-standard-procedure.html

**Comment by Dilbert66** [comment]
Unfortunately this is not a DSC alarm system.  The firmware does not support this system.

**Reply by younesetre** [reply]
Too bad, one adventure less.
Thanks lot

**Comment by albaboun** [comment]
hi dillbert66 , thank you for incredible proyect

I dont know how to send commandas from HA to dsc.  ( example ; chime on off) can you give any advice ?

thank you very much

**Comment by Dilbert66** [comment]
Use  the service esphome.xxxxx_alarm_keypress (where xxxx is the name you gave to your alarm panel in the yaml config)  you can send the cmd "C" or alternatively send the cmd "*4" in parameter "keys"  to toggle the chime.  You can send any valid dsc keypresses this way.

**Comment by albaboun** [comment]
Thank You very Much!!
Saludos,


Alberto Baboun

**Comment by albaboun** [comment]
hi dillbert66,

i cant figure out how to trigger an automation when alarm is fired. thank you for your help

**Reply by Dilbert66** [reply]
Look for the partition status sensor.dscalarm_partition_1_status to show "triggered" then you can have HA do an action. For more info on actions, you might want to look into the HA docs.

**Comment by Seamus-Mccartin** [comment]
Hi Dilbert66,

Is there a timeline on when the features in the other branches, mainly the keypad branch, are going to be merged into master and documented? I could probably puzzle it out over a weekend if I wanted to but a guide would be helpful.

**Comment by Dilbert66** [comment]
No timeline at this point since this is one of many projects I do  in my spare time.  I'm pretty busy with my paying gigs right now so havent had a chance to go back and work on it.  The "keypad" branch is obsolete anyhow so I've removed it. I'm doing most testing on the "new" branch right now which will move to dev when I'm done.  Most of the work really is involving creating a full LCD keypad emulation in firmware that matches a physical lcd with similar menu's. I'm about 80% there.  There really isnt much new  for documenation anyhow as the same functions will be there except the keypad card will now work more like a real LCD for control.  

Is there something in particular that you are wanting or not sure about?

**Reply by ikeserpa** [reply]
Hi Alain, I am very grateful for the excellent work you have done with this project. I built the board and I'm already using the resources in HA. I would be immensely grateful if you could guide me to put a keyboard on the HA lovelace. Do you have a guide that can help me with this? Thank you very much.

**Comment by whiteas2** [comment]
Hi. Thank you for the great project. I have built the interface and was able to connect to a test alarm panel and read zone status. When I connect to my installed alarm I get a keybus error on the existing keypads if the esp device is attached. The installed alarm has multiple keypads and expanders. Any idea what the problem could be?

**Comment by olddawgpowers** [comment]
Very nice project.  I currently have Envisalink and it works quite well, but would like to test this as an alternative and/or backup.  Is there any issue with using this and Envisalink?

**Comment by Dilbert66** [comment]
No issues at all. I have both running on the same system.

**Reply by olddawgpowers** [reply]
Thanks for fast response.

**Comment by olddawgpowers** [comment]
Great job on this project Alain. Had a Nodemcu ESP8266, some P817 Optos so I gave this a try to run parallel with my Envisalink.   I used the “new” version.   Very impressive and  fast response.

**Comment by Dilbert66** [comment]
Thanks. Glad you got it working.  For best operation though an ESP32 is recommended as the ESP8266 is tight on resources with the latest code.

**Comment by olddawgpowers** [comment]
For others that are using Envisalink.  I have been using Envisalink for many years and use EyezOn monitoring.  It has been solid with no false alarm reports.   I started using Envisalink Home Assistant integration a couple of years ago and would notice occasional states not updating.  I thought it was a zone switch or possibly the PowerSeries zone wiring.  I installed the esphome-dsckeybus and ran them parallel it appears the Envislink integration is not updating correctly.

![Screenshot 2022-10-16 131841](https://user-images.githubusercontent.com/36567590/196052176-1fe33ae2-80fd-406a-8498-8429546675e2.png)

In the attached image the dscalarm Garage Door(z2) is the esphome device I added and the Garage Door is the Envisalink. The esphome device reports correctly showing both times I open and closed the door at 2:55 and 3:01. The Envisalink shows the first open and close but almost immediately reports a "false" open. At approximately 3:01 I open and close the door again and it reports correctly closed.

**Comment by JacoAnnandale** [comment]
Good day

Thank you for a very well written implementation of the DSC Alarm. Works like a charm.

Sometimes I need to bypass a couple of zones, is there a way, with the help of your code, to bypass a group of zones? or just one zone will also work.

Thanks

**Comment by Dilbert66** [comment]
Just use the usual dsc keypad cmds.  My code is basically a keypad.  Send *1xx# where xx is the zone or if you want to bypass all zones, send *199#

**Reply by JacoAnnandale** [reply]
Thanks, will give it a go !!

**Comment by jet202020** [comment]
Hello Gilbert,
Amazing project development. Thanks.
My panel is PC1864. At the beginning, it was impossible to download the code to the ESP32Dev, error message were :

INFO Successfully compiled program.
INFO Resolving IP address of dsc-link.local in mDNS
INFO Resolving IP address of dsc-link.local
ERROR Error resolving IP address of dsc-link.local. Is it connected to WiFi?
ERROR (If this error persists, please set a static IP address: [https://esphome.io/components/wifi.html#manual-ips](https://www.google.com/url?sa=E&q=https%3A%2F%2Fesphome.io%2Fcomponents%2Fwifi.html%23manual-ips))
ERROR Error resolving IP address: Error resolving address with mDNS: Did not respond. Maybe the device is offline., [Errno -5] No address associated with hostname

I did several test, until I decide to eliminate modules to see which one was causing the trouble, and it was the SWITCH seccion. Eliminating this parts, the ESP board was able to be program by USB and OTA.

Any clue or suggestions? 
Is the SWITCH module necessary?

**Reply by Dilbert66** [reply]
I'm very surprised the switch component would give you that error. It has nothing to do with ip resolving.  I suspect it was just coincidental and just related to mdns resolving.  Sure, you can remove it  but you might need to comment out a couple places that reference it.

**Comment by jet202020** [comment]
Alain, what do you suggest to do to include again de "Switch" rutine?

Regards,
Ricardo.

**Reply by Dilbert66** [reply]
I'm not sure I understand your question? Just put it back? It's not a critical component so you can test without it.

**Comment by jet202020** [comment]
It works!
The example card you suppied, must go on a new folder inside the www folder?

**Reply by Dilbert66** [reply]
Either a sub folder or directly inside. If using a sub folder, you need to add it to the path in the resources config.

**Reply by jet202020** [reply]
Hi. Dilbert,
following your instruction and codes, the dashboard DSC Panel is working partially. The key are responding, but there is not message display , visual signs or sound on it. May you have a suggestion or clue, what i am doing bad?
Changes of status, open zones, connections, are visible in the Overview dashboard.

**Reply by Dilbert66** [reply]
Impossible for me to guess what it is that you are doing.  Just ensure that the sensors you used for the display lines are correct for your setup.  Don't assume the sensor names used in the example are the same as yours.

**Reply by jet202020** [reply]
Dilbert, which code should I match the sensors name against which code?  I changed names and types adapted to sensors I have installed.

**Reply by jet202020** [reply]
Ready. The entity elements were updaated and matched. Now the panel shows the info.
New question: i would like to reach the Bypass Zone information as Entity. How can i create/implement a entity reading that value? 
Thx

**Comment by jet202020** [comment]
Was fixed. Problem was the matches between entities.
Thanks and Regards.
And again, thanks by sharing your work. It will give me a secure feeling
because more alarm control and know status.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Libre
de virus.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
